### DISCUSSION TITLE
How to  make  'IO Peripherals Telnet'  component  work properly ?

### DISCUSSION THREAD

**Discussion by Jeffer-Lin** [hneemann/Digital#731] [Q&A] [answered]
I have git fetch  **telnet branch**  222269ef3 ("fixes some minor issues, see #726", 2021-04-29)
I draw a telnet/kbd to terminal test.dig   (at attached file)

dig file  : Telnet port set '**2323**'  , **'check**'  commands box
#**telnet localhost 2323**
Trying 127.0.0.1...
Connected to localhost.
Escape character is '^]'.
Type anthing is no response

dig file : Telnet port set '**2323**'  , '**uncheck**'  commands box
In os command terminal ,  #**telnet local 2323**
type something , it just echo back ,
I only succed one time , but not full function, The simulator terminal get partial character I had typed.
**How can I properly let the Telnet componet work ?**

[keyboard_telnet_terminal_break .dig.txt](https://github.com/hneemann/Digital/files/6402691/keyboard_telnet_terminal_break.dig.txt)

**Comment by hneemann** [comment, marked as answer]
This is a race condition. When rd is high, the last received character is output to out. However, the rising edge of the clock deletes the last character when rd is high.
In your circuit, however, rd is always high. It may be that the character is received and output to out, and there is not enough time to process the character until the next rising edge of the clock.
